contracts/zero-ex/test/protocol_fees_test.ts

```diff
@@ -35,41 +38,128 @@ blockchainTests.resets('ProtocolFees', env => {
         await weth.approve(protocolFees.address, constants.ONE_ETHER).awaitTransactionSuccessAsync({ from: payer });
     });
 
+    function collect(
+        poolId: string,
+        amount: BigNumber,
+        etherValue: BigNumber,
+    ): PromiseWithTransactionHash<TransactionReceiptWithDecodedLogs> {
+        return protocolFees
+            .collectProtocolFee(poolId, amount, weth.address)
+            .awaitTransactionSuccessAsync({ from: payer, value: etherValue });
+    }
+
+    function transferFees(poolId: string): PromiseWithTransactionHash<TransactionReceiptWithDecodedLogs> {
+        return protocolFees.transferFeesForPool(poolId, staking.address, weth.address).awaitTransactionSuccessAsync();
+    }
+
+    describe('FeeCollector', () => {
+        it('should disallow unauthorized initialization', async () => {
+            const pool = hexUtils.random();
+
+            await collect(pool, constants.ONE_ETHER, constants.ZERO_AMOUNT);
+            await transferFees(pool);
+
+            const feeCollector = new FeeCollectorContract(
+                await protocolFees.getFeeCollector(pool).callAsync(),
+                env.provider,
+                env.txDefaults,
+            );
+
+            const tx = feeCollector
+                .initialize(weth.address, staking.address, pool)
+                .sendTransactionAsync({ from: unauthorized });
+            expect(tx).to.revertWith(new AuthorizableRevertErrors.SenderNotAuthorizedError(unauthorized));
```

Comment:
```suggestion
            return expect(tx).to.revertWith(new AuthorizableRevertErrors.SenderNotAuthorizedError(unauthorized));
```
`expect(tx).to.reverWith(...)` is async, so we should either await or return the promise. Our convention is to `return`.